File: leetcode/go/412-fizz-buzz.go
package hackerrank

func fizzBuzz(n int) []string {
    response := make([]string, 0)
    for i := 1; i <= n; i++ {
        if i % 3 == 0 && i % 5 == 0 {
            response = append(response, "FizzBuzz")
        } else if i % 3 == 0 {
            response = append(response, "Fizz")
        } else if i % 5 == 0 {
            response = append(response, "Buzz")
        }else {
            response = append(response, strconv.Itoa(i))
        }
    }
    return response
}

File: leetcode/rust/412-fizz_buzz.rs
pub fn fizz_buzz(n: i32) -> Vec<String> {
    let mut result = Vec::new();
    for i in 1..n + 1 {
        match (i % 3 == 0, i % 5 == 0) {
            (true, true) => result.push("FizzBuzz".to_string()),
            (true, false) => result.push("Fizz".to_string()),
            (false, true) => result.push("Buzz".to_string()),
            (false, false) => result.push(i.to_string())
        }
    }
    return result;
}

#[test]
fn test_fizz_buzz() {
    assert_eq!(fizz_buzz(15), vec!["1", "2", "Fizz", "4", "Buzz", "Fizz", "7", "8", "Fizz", "Buzz", "11", "Fizz", "13", "14", "FizzBuzz"]);
}